### ISSUE TITLE
Declarative node collection validation is ignored

### ISSUE THREAD

**Issue by feelgood-interface** [n8n-io/n8n#11260]
### Bug Description

`parameterValue` will not be the collection's property values as object, but rather the specific collection's property value.
https://github.com/n8n-io/n8n/blob/26ad091f473bca4e5d3bdc257e0818be02e52db5/packages/core/src/NodeExecuteFunctions.ts#L2374

Therefore, the validation will be ignored.
https://github.com/n8n-io/n8n/blob/26ad091f473bca4e5d3bdc257e0818be02e52db5/packages/core/src/NodeExecuteFunctions.ts#L2330

### To Reproduce

```
{
	displayName: 'Additional Fields',
	name: 'additionalFields',
	default: {},
	options: [
		{
			displayName: 'Number',
			name: 'number',
			default: 0,
			routing: {
				send: {
					type: 'body',
					property: 'number',
					value: '={{$value}}',
				},
			},
			type: 'number',
			validateType: 'number',
		},
		{
			displayName: 'Option',
			name: 'option',
			default: 'option1',
			options: [
				{
					name: 'Option 1',
					value: 'option1',
				},
				{
					name: 'Option 2',
					value: 'option2',
				},
			],
			routing: {
				send: {
					type: 'body',
					property: 'option',
					value: '={{$value}}',
				},
			},
			type: 'options',
			validateType: 'options',
		},
	],
	placeholder: 'Add Field',
	type: 'collection',
}
```

### Expected behavior

Collection's properties need to be validated.

### Operating System

Ubuntu 24

### n8n Version

1.62

### Node.js Version

20

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @feelgood-interface,

We have created an internal ticket to look into this which we will be tracking as "GHC-319"

**Comment by Joffcom** [comment]
@feelgood-interface did you also put in a PR for this?

**Comment by feelgood-interface** [comment]
@Joffcom no. This PR is just for the root property validation https://github.com/n8n-io/n8n/pull/11259

**Comment by Joffcom** [comment]
Ah ok in that case I will get this one moved internally for prioritisation :)

**Comment by Joffcom** [comment]
We’re taking some time to review and tidy up our issue tracker. It looks like this issue hasn’t had any recent activity for the past 6 months. To keep our backlog focused on current priorities, we’re closing this issue for now.

That said, this doesn’t mean the issue is forgotten! If this is still relevant or you have more information to add, feel free to reopen it or create a new issue and we’d be happy to revisit it.

Thank you for understanding and for contributing to the project!
